std/src/main/java/dk/gtz/graphedit/plugins/syntaxes/petrinet/importing/tapaal/TapaalPNMLExporter.java

```diff
@@ -0,0 +1,126 @@
+package dk.gtz.graphedit.plugins.syntaxes.petrinet.importing.tapaal;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+
+import dk.gtz.graphedit.exceptions.ExportException;
+import dk.gtz.graphedit.model.ModelProjectResource;
+import dk.gtz.graphedit.plugins.syntaxes.petrinet.model.ModelArc;
+import dk.gtz.graphedit.plugins.syntaxes.petrinet.model.ModelPlace;
+import dk.gtz.graphedit.plugins.syntaxes.petrinet.model.ModelTransition;
+import dk.gtz.graphedit.spi.IExporter;
+
+public class TapaalPNMLExporter implements IExporter {
+    private final static Logger logger = LoggerFactory.getLogger(TapaalPNMLExporter.class);
+    private final XmlMapper xmlMapper;
+    private final String syntaxName;
+
+    public TapaalPNMLExporter(String syntaxName) {
+	this.syntaxName = syntaxName;
+	this.xmlMapper = new XmlMapper();
+	this.xmlMapper.registerModule(new Jdk8Module());
+    }
+
+    @Override
+    public String getName() {
+	return "tapaal";
+    }
+
+    @Override
+    public void exportFile(ModelProjectResource resource, Path newFilePath) throws ExportException {
+	var syntaxName = Optional.ofNullable(resource.metadata().get("graphedit_syntax"));
+	if(!syntaxName.isPresent())
+	    throw new ExportException("No syntax specified for resource");
+	if(!syntaxName.get().equals(this.syntaxName))
+	    throw new ExportException("Invalid syntax for TapaalPNMLExporter '%s' (expected '%s')".formatted(syntaxName.get(), syntaxName));
+	var pnml = fromResource(resource);
+	try {
+	    xmlMapper.writeValue(newFilePath.toFile(), pnml);
+	} catch (IOException e) {
+	    throw new ExportException("Failed to write PNML to file", e);
+	}
+    }
+
+    private PNML fromResource(ModelProjectResource resource) throws ExportException {
```

Comment:
Make sure to test if this exports tapaal compatible pnml’s. At time of writing, I havent actually tested it yet.